runtime/array.c

```diff
@@ -661,14 +667,14 @@ CAMLprim value caml_array_concat(value al)
       caml_raise_out_of_memory();
     }
   }
-  /* Build the parameters to caml_array_gather */
+  /* Build the parameters for the [gather] function. */
```

Comment:
This comment refers to line 657/663 above: shouldn't it be `n * sizeof(intnat)` (unrelated to this PR).

Reply:
Indeed, I pushed this in a separate commit.